### PR TITLE
feat: Add UDT to `avm/res/operational-insights/workspace`

### LINKED ISSUE #3798: [Failed pipeline] avm.res.operational-insights.workspace
Failed run: https://github.com/Azure/bicep-registry-modules/actions/runs/11854921204

### PR DESCRIPTION
**Pull request by krbar**
## Description

Add User Defined Types to the Log Analytics Workspace module.
Also fixes #3798 

## Pipeline Reference

| Pipeline |
| -------- |
|      [avm.res.operational-insights.workspace](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.operational-insights.workspace.yml)    |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [x] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/operational-insights/workspace/data-export/main.bicep
Comment (AlexanderSehr) [inline avm/res/operational-insights/workspace/data-export/main.bicep:63]: Just in case you did not know - this works too 😄 

```suggestion
@description('A type for a data export destination.')
type destinationType = {
```

File: avm/res/operational-insights/workspace/data-export/main.bicep
Comment (AlexanderSehr) [inline avm/res/operational-insights/workspace/data-export/main.bicep:74]:
```suggestion
  metaData: {
    @description('Optional. Allows to define an Event Hub name. Not applicable when destination is Storage Account.')
    eventHubName: string?
  }?
}
```

If only used in one location why 2 types?

File: avm/res/operational-insights/workspace/data-export/main.bicep
Comment (AlexanderSehr) [reply avm/res/operational-insights/workspace/data-export/main.bicep:63]: If you want to adopt this suggestion, the same comment applies to all other types. If not, nevermind. But I think there'll be a spec eventually 😄

Comment (krbar) [reply avm/res/operational-insights/workspace/data-export/main.bicep:64]: Thanks! No, I didn't know it, I will add the descriptions.

File: avm/res/operational-insights/workspace/data-export/main.bicep
Comment (krbar) [reply avm/res/operational-insights/workspace/data-export/main.bicep]: I merged these two types into one, but somehow I overlooked your commitable suggestion.... 🤷‍♂️

**Comment by jtracey93** [comment]
Hey @krbar ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**
